### ISSUE TITLE
rocm-opencl-runtime: make can't find elfio/elf_types.hpp

### ISSUE THREAD

**Issue by joeftiger** [rocm-arch/rocm-arch#457]
When trying to update `rocm-opencl-runtime` to `3.9.0-1`, it fails to find `elfio/elf-types.hpp`. I made sure that all other packages are updated and that the path `/opt/rocm/rocclr/include/elf/elfio/elf-types.hpp` exists.

<details><summary>Building</summary>
<p>
_Preparation ommitted_

```
-- Configuring done
-- Generating done
-- Build files have been written to: /home/julius/pamac-tmp/pamac-build/rocm-opencl-runtime/src
Scanning dependencies of target OpenCL
[  4%] Building C object khronos/icd/CMakeFiles/OpenCL.dir/loader/icd_dispatch.c.o
[  8%] Building C object khronos/icd/CMakeFiles/OpenCL.dir/loader/linux/icd_linux.c.o
[  8%] Building C object khronos/icd/CMakeFiles/OpenCL.dir/loader/icd.c.o
Scanning dependencies of target Common
[ 12%] Building C object khronos/icd/CMakeFiles/OpenCL.dir/loader/linux/icd_linux_envvars.c.o
[ 12%] Building CXX object tests/ocltst/module/common/CMakeFiles/Common.dir/BaseTestImp.cpp.o
[ 16%] Building CXX object tests/ocltst/module/common/CMakeFiles/Common.dir/OCLTestImp.cpp.o
Scanning dependencies of target amdocl64
[ 16%] Building CXX object tests/ocltst/module/common/CMakeFiles/Common.dir/OCLTestListImp.cpp.o
[ 20%] Building CXX object tests/ocltst/module/common/CMakeFiles/Common.dir/OCLTestUtils.cpp.o
[ 25%] Building CXX object amdocl/CMakeFiles/amdocl64.dir/cl_memobj.cpp.o
[ 25%] Building CXX object tests/ocltst/module/common/CMakeFiles/Common.dir/OCLThread.cpp.o
[ 25%] Building CXX object amdocl/CMakeFiles/amdocl64.dir/cl_program.cpp.o
[ 29%] Building CXX object amdocl/CMakeFiles/amdocl64.dir/cl_sdi_amd.cpp.o
[ 33%] Building CXX object tests/ocltst/module/common/CMakeFiles/Common.dir/OCLWrapper.cpp.o
In file included from /opt/rocm/rocclr/include/elf/elf.hpp:31,
                 from /opt/rocm/rocclr/include/device/device.hpp:32,
                 from /opt/rocm/rocclr/include/platform/context.hpp:25,
                 from /opt/rocm/rocclr/include/platform/command.hpp:36,
                 from /opt/rocm/rocclr/include/vdi_common.hpp:26,
                 from /home/julius/pamac-tmp/pamac-build/rocm-opencl-runtime/src/ROCm-OpenCL-Runtime-rocm-3.9.0/amdocl/cl_common.hpp:32,
                 from /home/julius/pamac-tmp/pamac-build/rocm-opencl-runtime/src/ROCm-OpenCL-Runtime-rocm-3.9.0/amdocl/cl_memobj.cpp:21:
/opt/rocm/rocclr/include/elf/elfio/elfio.hpp:42:10: fatal error: elfio/elf_types.hpp: No such file or directory
   42 | #include <elfio/elf_types.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~
compilation terminated.
/home/julius/pamac-tmp/pamac-build/rocm-opencl-runtime/src/ROCm-OpenCL-Runtime-rocm-3.9.0/tests/ocltst/module/common/OCLWrapper.cpp: In member function ‘_cl_command_queue* OCLWrapper::clCreateCommandQueue(cl_context, cl_device_id, cl_command_queue_properties, cl_int*)’:
/home/julius/pamac-tmp/pamac-build/rocm-opencl-runtime/src/ROCm-OpenCL-Runtime-rocm-3.9.0/tests/ocltst/module/common/OCLWrapper.cpp:184:14: warning: ‘_cl_command_queue* clCreateCommandQueue(cl_context, cl_device_id, cl_command_queue_properties, cl_int*)’ is deprecated [-Wdeprecated-declarations]
  184 |     return ::clCreateCommandQueue(context, device, properties, errcode_ret);
      |              ^~~~~~~~~~~~~~~~~~~~
make[2]: *** [amdocl/CMakeFiles/amdocl64.dir/build.make:82: amdocl/CMakeFiles/amdocl64.dir/cl_memobj.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /home/julius/pamac-tmp/pamac-build/rocm-opencl-runtime/src/ROCm-OpenCL-Runtime-rocm-3.9.0/tests/ocltst/include/OCLWrapper.h:26,
                 from /home/julius/pamac-tmp/pamac-build/rocm-opencl-runtime/src/ROCm-OpenCL-Runtime-rocm-3.9.0/tests/ocltst/module/common/OCLWrapper.cpp:21:
/home/julius/pamac-tmp/pamac-build/rocm-opencl-runtime/src/ROCm-OpenCL-Runtime-rocm-3.9.0/khronos/headers/opencl2.2/CL/cl.h:1813:1: note: declared here
 1813 | clCreateCommandQueue(cl_context                     context,
      | ^~~~~~~~~~~~~~~~~~~~
/home/julius/pamac-tmp/pamac-build/rocm-opencl-runtime/src/ROCm-OpenCL-Runtime-rocm-3.9.0/tests/ocltst/module/common/OCLWrapper.cpp:184:75: warning: ‘_cl_command_queue* clCreateCommandQueue(cl_context, cl_device_id, cl_command_queue_properties, cl_int*)’ is deprecated [-Wdeprecated-declarations]
  184 |     return ::clCreateCommandQueue(context, device, properties, errcode_ret);
      |                                                                           ^
In file included from /home/julius/pamac-tmp/pamac-build/rocm-opencl-runtime/src/ROCm-OpenCL-Runtime-rocm-3.9.0/tests/ocltst/include/OCLWrapper.h:26,
                 from /home/julius/pamac-tmp/pamac-build/rocm-opencl-runtime/src/ROCm-OpenCL-Runtime-rocm-3.9.0/tests/ocltst/module/common/OCLWrapper.cpp:21:
/home/julius/pamac-tmp/pamac-build/rocm-opencl-runtime/src/ROCm-OpenCL-Runtime-rocm-3.9.0/khronos/headers/opencl2.2/CL/cl.h:1813:1: note: declared here
 1813 | clCreateCommandQueue(cl_context                     context,
      | ^~~~~~~~~~~~~~~~~~~~
In file included from /opt/rocm/rocclr/include/elf/elf.hpp:31,
                 from /opt/rocm/rocclr/include/device/device.hpp:32,
                 from /opt/rocm/rocclr/include/platform/context.hpp:25,
                 from /opt/rocm/rocclr/include/platform/command.hpp:36,
                 from /opt/rocm/rocclr/include/vdi_common.hpp:26,
                 from /home/julius/pamac-tmp/pamac-build/rocm-opencl-runtime/src/ROCm-OpenCL-Runtime-rocm-3.9.0/amdocl/cl_common.hpp:32,
                 from /home/julius/pamac-tmp/pamac-build/rocm-opencl-runtime/src/ROCm-OpenCL-Runtime-rocm-3.9.0/amdocl/cl_program.cpp:21:
/opt/rocm/rocclr/include/elf/elfio/elfio.hpp:42:10: fatal error: elfio/elf_types.hpp: No such file or directory
   42 | #include <elfio/elf_types.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~
compilation terminated.
[ 33%] Building CXX object tests/ocltst/module/common/CMakeFiles/Common.dir/Timer.cpp.o
make[2]: *** [amdocl/CMakeFiles/amdocl64.dir/build.make:95: amdocl/CMakeFiles/amdocl64.dir/cl_program.cpp.o] Error 1
[ 37%] Building CXX object tests/ocltst/module/common/CMakeFiles/Common.dir/OCLGLCommon.cpp.o
In file included from /opt/rocm/rocclr/include/elf/elf.hpp:31,
                 from /opt/rocm/rocclr/include/device/device.hpp:32,
                 from /opt/rocm/rocclr/include/platform/context.hpp:25,
                 from /opt/rocm/rocclr/include/platform/command.hpp:36,
                 from /opt/rocm/rocclr/include/vdi_common.hpp:26,
                 from /home/julius/pamac-tmp/pamac-build/rocm-opencl-runtime/src/ROCm-OpenCL-Runtime-rocm-3.9.0/amdocl/cl_common.hpp:32,
                 from /home/julius/pamac-tmp/pamac-build/rocm-opencl-runtime/src/ROCm-OpenCL-Runtime-rocm-3.9.0/amdocl/cl_sdi_amd.cpp:21:
/opt/rocm/rocclr/include/elf/elfio/elfio.hpp:42:10: fatal error: elfio/elf_types.hpp: No such file or directory
   42 | #include <elfio/elf_types.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [amdocl/CMakeFiles/amdocl64.dir/build.make:108: amdocl/CMakeFiles/amdocl64.dir/cl_sdi_amd.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:557: amdocl/CMakeFiles/amdocl64.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 37%] Building CXX object tests/ocltst/module/common/CMakeFiles/Common.dir/OCLGLCommonLinux.cpp.o
[ 37%] Linking C shared library libOpenCL.so
[ 37%] Built target OpenCL
[ 37%] Built target Common
make: *** [Makefile:149: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
```
</p>
</details>

**Comment by tpkessler** [comment]
Hey @joeftiger, this is fixed with #455 that should be merged soon.

**Comment by acxz** [comment]
just for book keeping, dup of #452
